### PR TITLE
RavenDB-21050 - When restoring from incremental backup with atomic guard, a tombstone can be handled before creating the document

### PR DESCRIPTION
**Pull request by shaharhikri**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21050/When-restoring-from-incremental-backup-with-atomic-guard-a-tombstone-can-be-handled-before-creating-the-document

### Additional description

To create a document with atomic guard we send a ClusterWideTransaction.

This is done in two steps:

The atomic guard is created in the cluster state machine
The database creates the document.
If we have a full backup with an atomic-guarded document and an incremental backup with a tombstone to this document
The actual creation of the document can be after the tombstone is applied.

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [x] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Review by karmeli87 — APPROVED** [review]

**Comment by karmeli87** [comment]
please rebase

**Comment by karmeli87** [comment]
should this go to 5.4?

**Comment by shaharhikri** [comment]
> should this go to 5.4?

https://github.com/ravendb/ravendb/pull/18339

**Review by grisha-kotler — APPROVED** [review]

File: test/SlowTests/Issues/RavenDB-21050.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+namespace SlowTests.Issues;
+
+public class RavenDB_21050 : RavenTestBase
+{
+    public RavenDB_21050(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    [RavenFact(RavenTestCategory.BackupExportImport)]
+    public async Task ClusterWideTransaction_WhenRestoreFromIncrementalBackupAfterStoreAndDelete_ShouldDeleteInTheDestination()
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB-21050.cs]: let's check it for sharded database as well

**Review by grisha-kotler — APPROVED** [review]

**Pull request merged**